src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private transient int size;
+    private transient Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
 
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize(table);
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static final int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private void addNode(int hash, K key, V value, int index) {
+        Node<K, V> newNode = new Node<>(hash, key, value, table[index]);
+        table[index] = newNode;
+        size++;
+    }
+
+    private static int indexFor(int hash, int length) {
+        return hash & (length - 1);
+    }
+
+    private void resize(Node<K, V>[] oldTable) {
+        int newCapacity = oldTable.length * 2;
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                int newIndex = indexFor(node.hash, newCapacity);
+
+                Node<K, V> nextNode = node.next;
+                node.next = newTable[newIndex];
+                newTable[newIndex] = node;
```

Comment:
In the `resize` method, ensure that the order of nodes is preserved when transferring nodes from the old table to the new table. This is important to maintain the correct order of elements in the hash map.